builder.nix

```diff
@@ -106,28 +117,32 @@ pkgs.stdenv.mkDerivation {
     bash perl python3 dtc
   ] ++ lib.optional (!lib.versionAtLeast release "21" && release != "snapshot") python2;
 
-  buildPhase = let
-    makeArgs = lib.concatStringsSep " " [
-      "-j$NIX_BUILD_CORES"
-      "DISABLED_SERVICES='${lib.concatStringsSep " " disabledServices}'"
-      "PACKAGES='${lib.concatStringsSep " " packages}'"
-      "PROFILE='${profile}'"
-      "SHELL='${pkgs.runtimeShell}'"
-      (lib.optionalString (files != null) "FILES='./files'")
-      (lib.optionalString (rootFsPartSize != null) "ROOTFS_PARTSIZE=${toString rootFsPartSize}")
-      (lib.optionalString (extraImageName != null) "EXTRA_IMAGE_NAME='${extraImageName}'")
-    ];
-
-  in lib.optionalString (lib.versionOlder release "19") ''
-    # Hack around broken check for gcc
-    touch staging_dir/host/.prereq-build
-  '' + lib.optionalString (files != null) ''
-      # copy files to avoid making etc read-only
-      cp -r --no-preserve=all ${files} files
-  '' + ''
-    make image ${makeArgs}
-  '';
 
+  buildFlags = [
+    ".DEFAULT_GOAL=image"
```

Comment:
OpenWRT's [Makefile ](https://github.com/openwrt/openwrt/blob/32e4c50d24d08570c6ae34d05922f1d9d879c0f2/target/imagebuilder/files/Makefile)doesn't have the default target, and nixpkgs' `buildPhase` [doesn't have](https://github.com/NixOS/nixpkgs/blob/74753257638e0858632599ca0f239631d35bda49/pkgs/stdenv/generic/setup.sh#L1326) the `buildTarget` or anything like that 😢  (But there's a proposal: https://github.com/NixOS/nixpkgs/issues/28047). 

So we have to manually specify the target, using this [special variable](https://www.gnu.org/software/make/manual/make.html#Special-Variables). We could also patch the Makefile here. Let me know if there's a preference.

Reply:
OpenWRT's [Makefile ](https://github.com/openwrt/openwrt/blob/32e4c50d24d08570c6ae34d05922f1d9d879c0f2/target/imagebuilder/files/Makefile)doesn't have the default target, and nixpkgs' `buildPhase` [doesn't have](https://github.com/NixOS/nixpkgs/blob/74753257638e0858632599ca0f239631d35bda49/pkgs/stdenv/generic/setup.sh#L1326) the `buildTarget` or anything like that 😢  (But there's a proposal: https://github.com/NixOS/nixpkgs/issues/28047). 

So we have to manually specify the target, using this [special variable](https://www.gnu.org/software/make/manual/make.html#Special-Variables). We could also patch the Makefile here. Let me know if there's a preference.